ldms/src/core/ldms.c

```diff
@@ -760,23 +760,22 @@ static void __destroy_set_no_lock(void *v)
 		 */
 		pthread_mutex_lock(&x->lock);
 		TAILQ_FOREACH(ctxt, &x->ctxt_list, link) {
-			ctxt = TAILQ_FIRST(&x->ctxt_list);
 			switch (ctxt->type) {
 			case LDMS_CONTEXT_LOOKUP_READ:
-				if (ctxt->lu_read.s)
+				if ((ctxt->lu_read.s == set) && ctxt->lu_read.s)
```

Comment:
Is the second part of this check necessary? isn't it just if (ctxt->lu_read.s == set)?